### PR TITLE
Support mapbox 6.0 and later and switch to $ReactNativeMapboxGLIOSVersion

### PR DESCRIPTION
**Pull request by mfazekas**
Deprecates:
```
ENV["REACT_NATIVE_MAPBOX_MAPBOX_IOS_VERSION"] = "6.2.1"
```

New version:
```
$ReactNativeMapboxGLIOSVersion = "6.2.1"
```

### CODE REVIEW

File: scripts/autogenerate.js [hunk 1]
```diff
@@ -21,7 +21,7 @@ function readIosVersion() {
     'react-native-mapbox-gl.podspec',
   );
   const lines = fs.readFileSync(podspecPath, 'utf8').split('\n');
-  const mapboxLineRegex = /^REACT_NATIVE_MAPBOX_MAPBOX_IOS_VERSION\s*=\s*.+'~>\s+(\d+\.\d+)'$/;
+  const mapboxLineRegex = /^REACT_NATIVE_MAPBOX_MAPBOX_IOS_VERSION.*||.+'~>\s+(\d+\.\d+)'$/;
```
Comment (ferdicus) [inline scripts/autogenerate.js:24]: Can you please explain this line a bit. 
In `react-native-mapbox-gl.podspec` we're switching everything from `REACT_NATIVE_MAPBOX_MAPBOX_IOS_VERSION` to `rnmbgl_ios_version`, however in here we still seem to look for the previous name within the podspec?

**Review by ferdicus — COMMENTED** [review]
Thanks 👍 

Are you also going to add some form of update into the changelog or do you think the deprecation warning is enough?

File: scripts/autogenerate.js (hunk 1 above)
Comment (mfazekas) [reply scripts/autogenerate.js]: Thanks, simplified in current version to `default_ios_mapbox_version`

**Comment by mfazekas** [comment]
> Thanks 👍
> 
> Are you also going to add some form of update into the changelog or do you think the deprecation warning is enough?

The env was in our changelog https://github.com/react-native-mapbox-gl/maps/blob/master/CHANGELOG.md#810rc2 i've added this deprecation not the that link as well.

**Comment by ferdicus** [comment]
> > Thanks 👍
> > Are you also going to add some form of update into the changelog or do you think the deprecation warning is enough?
> 
> The env was in our changelog https://github.com/react-native-mapbox-gl/maps/blob/master/CHANGELOG.md#810rc2 i've added this deprecation not the that link as well.

Cool 👏 

Should we also add the `.netrc` info to the iOS installation guide?

**Comment by systemlevel** [comment]
Screen Shot 2020-10-02 at 1 22 44 PM

Been having a bit of trouble understanding how to approach this for this PR. Looks like it might be attempting to download the correct SDK from Mapbox but I'm unauthorized. Do we place the auth token somewhere in the podfile or elsewhere?

**Comment by mfazekas** [comment]
@systemlevel pls see https://docs.mapbox.com/ios/maps/overview/#configure-credentials on how to configure `.netrc` file

**Comment by systemlevel** [comment]
@mfazekas Thank you!

**Comment by systemlevel** [comment]
@mfazekas success with the `.netrc` file. However, any iOS SDK over 6.0.0 I'm getting a crash immediately when the app loads. 

```
Dyld Error Message:

dyld: Using shared cache: 64C38D35-50E5-3FF4-A607-6D00F0234183

Library not loaded: @rpath/MapboxMobileEvents.framework/MapboxMobileEvents

Referenced from: /Users/USER/Library/Developer/CoreSimulator/Devices/16EE9272-941F-4B44-B6BB-1A500A55102D/data/Containers/Bundle/Application/0037690B-5056-4C48-8B50-5AD737820C11/MAPAPP.app/Frameworks/Mapbox.framework/Mapbox

`**Reason: image not found**`
```

Perhaps https://github.com/react-native-mapbox-gl/maps/pull/1052 assists with this potentially?

**Comment by systemlevel** [comment]
@mfazekas https://github.com/react-native-mapbox-gl/maps/blob/65cab1c7f43deb2194dac95141529d1f610884c4/react-native-mapbox-gl.podspec 
That podspec update fixed it for me. iOS SDK 6.2.1 is working now. 

Thank you!

**Comment by melvinmt** [comment]
@mfazekas 

can you explain where/how I can set the `$ReactNativeMapboxGLIOSVersion` variable?

**Comment by ferdicus** [comment]
@melvinmt, it is described here: https://github.com/react-native-mapbox-gl/maps/pull/940

**Comment by melvinmt** [comment]
@ferdicus great, that worked, thank you!

**Pull request merged**